### DISCUSSION TITLE
Editing issues on pantry

### DISCUSSION THREAD

**Discussion by mrdummy228** [imRohan/Pantry#147] [Q&A] [answered]
Hello! Your service is most great i found out there, but there are 2 problems.
First, if you try to remove the placeholder "key": "value" pair and click save, and then open the basket again, that pair would still be here.
Second, which is related to first, you can't edit your pantry ANYHOW except by using the API itself, which causes discomfort.
Is it intended? If it is, then it's a huge mistake and should be fixed (my opinion).

**Comment by imRohan** [comment, marked as answer]
Great questions!

For the first issue, the dashboard should update the basket using a `POST` request, but is currently using `PUT`. I like your idea, and it makes sense to perform destructive actions via the dashboard. I will fix this.

As for the second issue, regarding editing the pantry itself, could you please provide more context? What are you trying to edit exactly?
This issue (https://github.com/imRohan/Pantry/issues/143) should address editing the Pantry name, but I'm curious to hear more about other items you're interested in editing.

**Reply by mrdummy228** [reply]
yes, i think it makes sense too. so for example my JSON basket is `{ "key": "value" }` (a dictionary), but i want it to be `[ {"key": "value"} ]` (dictionaries inside a list). i would be fine to edit it with using API, just i thought like you that not able to be editing your basket with dashboard just makes no sense.

**Reply by mrdummy228** [reply]
for example, i actually use your service to make up a JSON "database" (despite JSON isn't for databases LOL), in a email service. what if `User0` spams mails to `User1`? it, of course would be better to apply some punishments first, warnings and stuff, but no, we don't talk about that here. what if `User0` just keeps spamming even after a lot of punishments? well, now at this point delete this account. and THERE, about editing goes speech. it would be of course easier to just open the dashboard and erase the user's account data, than rather writing a program, running it (i noticed that doing changes to your dashboard via API also takes quite a time).

**Reply by mrdummy228** [reply]
and trying to set basket's value to [] with using API won't work. it just sets to {}

**Reply by imRohan** [reply]
You can't assign an array to a basket. The basket contents must be an object, which is why you are unable to save the structure outlined above.

To help, perhaps you could create a new basket for each user, that way you can CRUD users individually without having to deal with arrays (lists).

**Reply by mrdummy228** [reply]
a new basket for each user won't help: what if there would be more than 100 users?

**Reply by imRohan** [reply]
@mrdummy228 Good point, and yes that would be a limitation. I am thinking of increasing that number in the future, but for now, once you have 100 active users, you will need to get creative and perhaps delete inactive users.

**Reply by mrdummy228** [reply]
alright, i solved the issue without basket for each user. emails are stored in a basket, example: {"email_unique_code": {"topic": "blah blah", content: "abcdef"}} and it works
